### ISSUE TITLE
Choice of document editor

### ISSUE THREAD

**Issue by adriatic** [aurelia-community/community-wide-collaboration#9]
[Hemingway](http://www.hemingwayapp.com/help.html)

**Comment by fmonroe** [comment]
Thanks for the research and the suggestions. I reviewed all the above and my conclusion is:
- We should use Gitbook with VSCode, using an individual (or very small team) of publishers to bypass the 5 collaborators limit.
Longer response: Here is my rationale (some of these items are more or less subjective):
1. We don't want to force collaborators to learn/use yet another tool (such as Docusaurus/Hemingway).
2. Markdown and VSCode are both well-known and therefore we are not asking our collaborators to learn new technologies/applications.
3. We cannot afford GitBook with its current pricing structure
4. Gitbook is well known and becoming an established standard
5. The small team of publishers can use Hemingway to insure the text is 'readable', remembering than English is not the main language of many readers of AUCS.

**Comment by adriatic** [comment]
@fmonroe - your selection and reasing matched mine as well. While at the moment we should consider our selection just as our selection, we should revisit this again, once we get other AUCS team members input (via the survey)

**Comment by ShawnTalbert** [comment]
I don't use VSCode  (I use Webstorm which is superior IMHO).

I assume you mean **G**itHub flavored markdown, not 'BitHub'?

Hemmingway doesn't seem to support markdown, at least in my quick try with the tool?

As long as we have a way to easily submit content to a _publisher_ it shouldn't matter what I use for an editor?

**Comment by fmonroe** [comment]
>  it shouldn't matter what I use for an editor?

That is my understanding and one of the reason I favor this solution/workflow.

**Comment by maroy1986** [comment]
I see you forget one. ReadTheDoc seems to be pretty straight forward to connect to a GitHub repo and then parse the markdown.

https://github.com/rtfd/readthedocs.org

So this might be another way to look at!

**Comment by ShawnTalbert** [comment]
ReadTheDocs is RST not Markdown syntax?

**Comment by adriatic** [comment]
@ShawnTalbert you are correct in your comment above and to explain why I just expanded the bullet point in the begining of this **[Requirements](https://github.com/aurelia-community/community-wide-collaboration/issues/9#issue-290275861)** section.

This is probably the most important requirement as we always need to view our effort as adding the missing data to Aurelia - not to run in parallel with Aurelia.

**Comment by maroy1986** [comment]
Whoops, I misread for ReadTheDoc, sorry. My bad, I was under the impression that it was using markdown too.

**Comment by ShawnTalbert** [comment]
Does this reduce the flow to simply the following?
1. write docs using Markdown syntax (create the markdown source however ye like)
2. submit the markdown file(s) to an appropriate repo (perhaps in a branch for 'publisher' review)

For myself, I'd probably stick with the markdown plugins for webstorm. Others can use VSCode if that's their preference. Webstorm markdown editing has the additional nifty feature of being programming language aware as well. This means if writing a TypeScript example it will actually typecheck inside the markdown document.

**Comment by adriatic** [comment]
@ShawnTalbert if we decide to go with your description then the flow is reduced indeed, relative to my original statement **[above](https://github.com/aurelia-community/community-wide-collaboration/issues/9#issue-290275861)** - as the requirement for the versioning gets dropped. 

I would be careless to support such decision without a good understanding of the ratio between the additional effort associated with using Docusaurus and the value of having documents that support versioning. After all our task of creating the documentation is not really different from creating the software and we should use the same workflows for both.

**Comment by ShawnTalbert** [comment]
Sorry, I just took for granted the notion that using a repo gets us versioning by design?

Is there something you mean by the term _versioning_ that can't be covered by the version control system?

**Comment by adriatic** [comment]
@ShawnTalbert - there might be _something_, which is a reason for my interest in Docusaurus. Here is a quote from their website:
> You can use the version script to cut a new documentation version based on the latest content in the docs folder. That specific set of documentation will then be preserved and accessible even as the documentation in the docs folder changes moving forward.

The app versioning is different from docs versioning, since the app is a "bound" collection of source files, which version is set by the release process. Keeping a given user manual in GitHub ensures versioning for each markdown file, but not for the manual itself (unless we publish it as Gitbook, which acts as a "bundler and server")

**Comment by obedm503** [comment]
I'll throw in [Penflip](https://www.penflip.com/) as a suggestion. It supports markdown, versioning based on sentence instead of line, works online or you're with favorite text editor, and it's free for open projects https://web.appstorm.net/reviews/office/penflip-reinvents-git-for-writers/

**Comment by adriatic** [comment]
@obedm503 seems like a great suggestion and in my own opinion we need to choose between penflip and docusaurus, with the likely outcome that we can leave the decision to each team member. I will make a test using docusaurus and penflip to verify that they are completely exchangeable - and post my findings here.

**Comment by adriatic** [comment]
### Testing docusaurus
While docusaurus seems to be designed based on the specification of our needs 😏, it is still a bit un-sophisticated and very **react**-ive; no surprise there, as it is created by a few Facebook team members. More importantly, it does not support the nested file structures (see **[this comment](https://docusaurus.io/docs/en/navigation.html)**)
> Docusaurus currently does not support documents in nested folders; only in a flatfile structure. We are looking into adding support for nested folders.

Given our ambitious plans, this would result with a catastrophic mess.

Despite some really cool features, I am voting **NO** - but see the actual **[sample site](https://github.com/aucs-publishers/test-docusaurus)** anyway.

**Comment by adriatic** [comment]
### Comments to **[`Testing docusaurus`](https://github.com/aurelia-community/community-wide-collaboration/issues/9#issuecomment-360523044)**

<put your comments below>
...

**Comment by adriatic** [comment]
### Testing Penflip

**Comment by fmonroe** [comment]
About Docusaurus 
As I could _not_ believe such a show stopper; just to confirm: from Docusaurus' `Discord`: Today at 1:23 PM
That's correct. All of the doc markdown files currently go in the /docs folder of your project. I'm not sure on the specifics for plans to support nesting, but in practice I can say this hasn't been an issue for the sites I've built.
The way the documentation is stored and the way it's presented are separate. The actual markdown files are in a flat structure but they can be grouped arbitrarily in the website.
See reasonml.github.io as an example:
Markdown: https://github.com/reasonml/reasonml.github.io/tree/source/docs
Multiple Groupings: https://reasonml.github.io/docs/en/quickstart-javascript.html
Dedicated pages for specific groupings: https://reasonml.github.io/docs/en/community.html
The file that describes the groupings: https://github.com/reasonml/reasonml.github.io/blob/source/website/sidebars.json

**Comment by adriatic** [comment]
@fmonroe - I am having difficulties parsing:

> As I could not believe such a show stopper; just to confirm: from Docusaurus' Discord: Today at 1:23 PM

my understanding of your **[post above](https://github.com/aurelia-community/community-wide-collaboration/issues/9#issuecomment-360559859)** is that you agree with my **[own assessment](https://github.com/aurelia-community/community-wide-collaboration/issues/9#issuecomment-360523044)**

**Comment by fmonroe** [comment]
You are right. I feel the inability to break down the document into subfolders in Docusaurus is a 'show stopper', meaning IMO it cannot be used for our project.
